sql/expression/matchagainst.go

```diff
@@ -363,17 +376,21 @@ func (expr *MatchAgainst) inNaturalLanguageMode(ctx *sql.Context, row sql.Row) (
 		rowCountRow := rowCountRows[0]
 
 		// Calculate the relevancy (partially based on an old MySQL implementation)
-		//TODO: use an actual algorithm with a good distribution, however we're focusing on correctly returned results for now
-		docCount := float32(docCountRow[len(docCountRow)-1].(uint64))
-		globalCount := float32(globalCountRow[len(globalCountRow)-1].(uint64))
-		uniqueWords := float32(rowCountRow[2].(uint64))
-		fp := docCount / globalCount
-		sp := 1 + 1/(1+0.115*uniqueWords)
-		accumulatedRelevancy += fp * sp
+		// https://web.archive.org/web/20220122170304/http://dev.mysql.com/doc/internals/en/full-text-search.html
+		globalCount := float64(globalCountRow[len(globalCountRow)-1].(uint64))
+		uniqueWords := float64(rowCountRow[2].(uint64))
+		base := math.Log(docCount) + 1
+		normFactor := uniqueWords / (1 + 0.115*uniqueWords)
+		globalMult := math.Log(float64(expr.parentRowCount)/globalCount) + 1
+		accumulatedRelevancy += float32(base * normFactor * globalMult)
```

Comment:
It doesn't, since we can easily calculate it. Also, the linked docs are for MyISAM, which returns a different result than InnoDB, so I just used it as a general guide rather than their exact implementation. This lets us change the algorithm later if we can ever find out InnoDB's actual formula.